src/core.jl

```diff
@@ -72,7 +75,11 @@ end
 device_index_from_zero(part::DevicePart{Int}) = "$(part.name):$(part.index-1)"
 device_index_from_zero(part::DevicePart) = "$(part.name):$(part.index)"
 
-struct Device
+struct NamedDevice <: AbstractDevice
```

Comment:
Note, I'm not sure about this particular thing. The reason I added it is that the TPU naming scheme `/job:tpu_worker/replica:0/task:0/device:TPU:0` wasn't compatible with the existing scheme (because of the last part having two colons). I don't know anything about TensforFlow device naming, so there's probably a better solution.

Reply:
It looks like TensorFlow has changed naming conventions at some point since I last looked at it and we should just modify the existing `DevicePart` to handle it, rather than adding a new device type. I'll push to this PR to do so, if you don't mind.